### DISCUSSION TITLE
What happened to SecondDerivativeImplicitEuler ?

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#17034] [Q&A Modules: General] [answered]
Hi,

I vaguely remember that there used to be a SecondDerivativeImplicitEuler.C Kernel in solid_mechanics before it was migrated. Is this somewhere? Was it removed?

Thanks,
John

**Comment by aeslaughter** [comment]
@sapitts Do you know this answer?

**Reply by sapitts** [reply]
Looks like that particular Kernel was deprecated and not migrated over to tensor mechanics. It could be that similar functionality exists elsewhere in MOOSE, but where I do not know. Perhaps @cbolisetti or @jasondhales might know?

**Reply by cbolisetti** [reply]
I don't know :( Maybe @bwspenc?

**Reply by mangerij** [reply]
Well it would be nice to have it back... I remember it worked correctly.

**Comment by hugary1995** [comment, marked as answer]
If you are talking about this one: [https://github.com/idaholab/moose/blob/mlpack-15154/modules/solid_mechanics/src/kernels/SecondDerivativeImplicitEuler.C](https://github.com/idaholab/moose/blob/mlpack-15154/modules/solid_mechanics/src/kernels/SecondDerivativeImplicitEuler.C) , this is now simply a special case of the InertialForce kernel.

The inertial force kernel uses NewmarkBeta time integrator. A special case is
- density = 1.
- eta = 0, alpha = 0. No damping.
- Gamma = 0.5, beta = 0.25. These are default parameters of NemarkBeta.

which reduces to second order implicit euler.

**Reply by mangerij** [reply]
excellent.. Good to know here.
